src/SprykerEco/Zed/CrefoPay/Persistence/CrefoPayRepository.php

```diff
@@ -0,0 +1,314 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Zed\CrefoPay\Persistence;
+
+use ArrayObject;
+use Generated\Shared\Transfer\PaymentCrefoPayOrderItemCollectionTransfer;
+use Generated\Shared\Transfer\PaymentCrefoPayOrderItemToCrefoPayApiLogTransfer;
+use Generated\Shared\Transfer\PaymentCrefoPayOrderItemToCrefoPayNotificationTransfer;
+use Generated\Shared\Transfer\PaymentCrefoPayOrderItemTransfer;
+use Generated\Shared\Transfer\PaymentCrefoPayTransfer;
+use Orm\Zed\CrefoPay\Persistence\SpyPaymentCrefoPayOrderItemQuery;
+use Orm\Zed\CrefoPay\Persistence\SpyPaymentCrefoPayOrderItemToCrefoPayApiLogQuery;
+use Orm\Zed\CrefoPay\Persistence\SpyPaymentCrefoPayOrderItemToCrefoPayNotificationQuery;
+use Orm\Zed\CrefoPay\Persistence\SpyPaymentCrefoPayQuery;
+use Propel\Runtime\Collection\ObjectCollection;
+use Spryker\Zed\Kernel\Persistence\AbstractRepository;
+use SprykerEco\Zed\CrefoPay\Persistence\Mapper\CrefoPayPersistenceMapper;
+
+/**
+ * @method \SprykerEco\Zed\CrefoPay\Persistence\CrefoPayPersistenceFactory getFactory()
+ */
+class CrefoPayRepository extends AbstractRepository implements CrefoPayRepositoryInterface
+{
+    /**
+     * @param int $fkSalesOrder
+     *
+     * @return \Generated\Shared\Transfer\PaymentCrefoPayTransfer
+     */
+    public function findPaymentCrefoPayByFkSalesOrder(int $fkSalesOrder): PaymentCrefoPayTransfer
```

Comment:
looks more like `get*` methods

Reply:
Methods in repository and reader are refactored.